contracts/core/Registry.sol

```diff
@@ -21,27 +36,75 @@ contract Registry {
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners
+    ) external returns (uint256) {
+
+        IdentityDetails memory identityDetails = IdentityDetails(
+            name,
+            metadata,
+            // ASK : should identityId be used as the hash for attestationAddr?
+            // cause project can't have same attestAddr across chains
+            generateAttestationAddr(name, msg.sender)
+        );
+
+        identities[identityId++] = identityDetails;
+
+        for(uint i = 0; i < _owners.length; i++) {
+            _grantRoles(_owners[i], identityId);
+        }
+        
+        emit IdentityCreated(identityId, name, metadata, identityDetails.attestationAddr);
+        
+        return identityId;
+    }
+
+    /// @notice Updates the name of the identity
+    /// @dev This will also change the attestation address, since it's a hash that includes the name
+    function updateIdentityName(uint _identityId, string memory _name) onlyRoles(identityId) external {
+        identities[_identityId].name = _name;
+        identities[_identityId].attestationAddr = generateAttestationAddr(_name, msg.sender);
+
+        emit IdentityNameUpdated(_identityId, _name);
     }
 
-    // updates the name of the identity
-    // note that this will also change the attestation address, since it's a hash that includes the name
-    function updateIdentityName(uint _identityId, string memory _name) external {
+    function changeCoreOwner(uint _identityId, address _newCoreOwner) external {
+        // ASK -> the actual owner cannot be tracked by Solady ROLES. 
+        // Would this be stored on IdentityDetails ?
 
+        // @todo check if msg sender is owner
+        // @todo check if new owner is not already owner
+        // @todo regenrate attestation address 
+        // @todo change owner
     }
 
     // @todo override changing owners (core owner, not contributors) to make sure it updates attestation address
 
     // update the metadata of the identity
     // checks ownership role first, probably separate roles for this vs using it (owner vs user?)
-    function updateIdentityMetadata(uint _identityId, string memory _metadata) external {
+    function updateIdentityMetadata(uint _identityId, Metadata memory _metadata) onlyRoles(identityId) external {
+        identities[_identityId].metadata = _metadata;
 
+        emit IdentityMetadataUpdated(_identityId, _metadata);
     }
 
-    // this will use solmate ROLES to check if the msg.sender is an owner of the project
-    // @todo figure out how to best represent project ownership and details
-    function isOwnerOfProject(uint _identityId, address _owner) external view returns (bool) {
-        
+    /// @notice Returns if the given address is an owner of the project
+    function isOwnerOfIdentity(uint _identityId, address _owner) external view returns (bool) {
```

Comment:
renaming this to `isOwnerOfIdentity` for consistency.
Also do we need this function ? Can't we just expect folks to use hasAnyRole directly ?

Reply:
We could. `hasAnyRole()` uses a bitmap for the roles and requires the caller to put together the role id so may be a bit tricky for users. I like the idea of having these helper functions that make it easy to call with just the identityId and the owner.